internal/distributed/querycoord/client/client.go

```diff
@@ -46,7 +46,7 @@ type Client struct {
 
 // NewClient creates a client for QueryCoord grpc call.
 func NewClient(ctx context.Context, metaRoot string, etcdCli *clientv3.Client) (*Client, error) {
```

Comment:
`metaRoot string, etcdCli *clientv3.Client` can be removed from params.

ditto with other similar functions in this PR.
Thanks for cleaning it up. The interface will be much cleaner.